modules/ingestor/src/service/format.rs

```diff
@@ -38,17 +38,37 @@ impl<'g> Format {
     }
     pub fn from_bytes(bytes: &[u8]) -> Result<Self, Error> {
         let checksum = checksum(bytes);
-        if serde_json::from_slice::<Vulnerability>(bytes).is_ok() {
-            Ok(Format::OSV { checksum })
-        } else if serde_json::from_slice::<Csaf>(bytes).is_ok() {
-            Ok(Format::CSAF { checksum })
-        } else if serde_json::from_slice::<CveRecord>(bytes).is_ok() {
-            Ok(Format::CVE { checksum })
-        } else {
-            Err(Error::UnsupportedFormat("unknown".into()))
+
+        let mut potential_errors = Vec::new();
+
+        match serde_json::from_slice::<osv::schema::Vulnerability>(bytes) {
+            Ok(_) => return Ok(Format::OSV { checksum }),
+            Err(e) => {
+                potential_errors.push(format!("if osv: {}", e));
+            }
+        }
+
+        match serde_json::from_slice::<Csaf>(bytes) {
+            Ok(_) => return Ok(Format::CSAF { checksum }),
+            Err(e) => {
+                potential_errors.push(format!("if csaf: {}", e));
+            }
         }
+
+        match serde_json::from_slice::<CveRecord>(bytes) {
+            Ok(_) => return Ok(Format::CVE { checksum }),
+            Err(e) => {
+                potential_errors.push(format!("if cve: {}", e));
+            }
+        }
+
+        Err(Error::UnsupportedFormat(format!(
+            "unknown :: {:?}",
+            potential_errors
+        )))
     }
```

Comment:
There's probably performance opportunities in here, e.g. use `RawValue` for everything in the doc except the field[s] required to recognize the format. That way, we'd save a crapton of memory allocation and we'd know exactly what format the errors applied to. Also, we should probably order them by most common expected.

We can do that later, obviously.

Reply:
pb https://github.com/trustification/trustify/issues/257